apollo-federation/src/sources/connect/validation/extended_type.rs

```diff
@@ -311,6 +341,55 @@ fn validate_field(
                 field: &field.name,
             },
         ));
+
+        let all_variables = json_selection
+            .as_ref()
+            .map(extract_params_from_selection)
+            .into_iter()
+            .flatten()
+            .chain(
+                body_selection
+                    .as_ref()
+                    .map(extract_params_from_selection)
+                    .into_iter()
+                    .flatten(),
+            )
+            .chain(
+                url_template
+                    .as_ref()
+                    .map(|(u, _)| u.variables().cloned())
+                    .into_iter()
+                    .flatten(),
+            )
+            .collect_vec();
```

Comment:
Should we be including headers here? I'm thinking an API might take a parameter as a header instead of in the URL path.

Reply:
definitely, good call. we need to update expansion for this too. this is a good reason to consolidate that code. nick and i came up with a plan yesterday

Reply:
this very valid request turned into a major exercise. i didn't want to replicate source/connect header merging logic, so i redid a bunch of stuff to allow using the Connector model struct in validations. once i did that, i was able to somewhat simplify the expansion code by moving `@key` generation into the model layer.